### DISCUSSION TITLE
Reverb G2 won't start with either OpenXR or SteamVR as active runtime. Quest 2 with Airlink crashes

### DISCUSSION THREAD

**Discussion by Braxusx** [fholger/crysis_vrmod#12] [General]
I tried running the new 0.2.0 release with OpenXR as the active runtime using my Reverb G2. This is without SteamVR running since it's a 
 WMR headset and the new 0.2.0 doesn't need it right? The game starts but won't hook into VR. 

I also tried using SteamVR and setting SteamVR as the active runtime, but with no success there either.

Then I tried running it on my Quest 2 using Airlink and it works fine, except I kept getting random crashes. 

Here are 4 separate game logs I renamed for each scenario. Hopefully one of them will help out with development. 

Thank you for working on this awesome mod!

P.S. Is openxr_loader.dll file needed ? Or can that be utilized like praydogs mods do? Just curious cause I noticed the last releases had the openvr dll file included.

[Game_Reverb G2 OpenXR_No_SteamVR.log](https://github.com/fholger/crysis_vrmod/files/11240715/Game_Reverb.G2.OpenXR_No_SteamVR.log)

[Game_Reverb G2 SteamVR with OpenXR as active runtime.log](https://github.com/fholger/crysis_vrmod/files/11240718/Game_Reverb.G2.SteamVR.with.OpenXR.as.active.runtime.log)

[Game Reverb G2 with SteamVR as active runtime (game freezes or hangs on start).log](https://github.com/fholger/crysis_vrmod/files/11240714/Game.Reverb.G2.with.SteamVR.as.active.runtime.game.freezes.or.hangs.on.start.log)

[Game_Quest 2 with Airlink Crash.log](https://github.com/fholger/crysis_vrmod/files/11240721/Game_Quest.2.with.Airlink.Crash.log)

**Comment by divxmaster1** [comment]
I have exactly the same issues as you with my G2.  Wont hook into VR, with the same error:
XR_ERROR_FUNCTION_UNSUPPORTED (-7)

Tried everything I can think of, even put fholgers CAS mod in there. The CAS mod does indeed hook, but still no go.

Divxmaster

**Reply by Braxusx** [reply]
Ahh I see. It might just be an issue to get sorted out since he switched from openvr to openxr. This is a pre-release, so maybe my logs above will help out a bit. If you can post your Game log as well.

**Comment by divxmaster1** [comment]
Have attached it.  I did notice something, we are running the same older version of openxr (v109.2111.23003).
I have forced installed that version, as it is the only one that allows fholgers 😄   CAS (vrsharpencolor version) to work with opencomposite.

I can switch versions of open with powershell and windows developer mode, might test that later.

[Game.log](https://github.com/fholger/crysis_vrmod/files/11300262/Game.log)

**Comment by divxmaster1** [comment]
I got some more time to look into this, tried the latest development build, still failed, but with different error:

OpenXR runtime: Windows Mixed Reality Runtime (v109.2111.23003)
Found vtable index 5 for hook ISystem::Render
Found vtable index 14 for hook ISystem::Quit
Found handle for D3D10 renderer, attempting to hook shadow rendering function
Retrieved swap chain: 194667712l
Acquiring device...
Game is rendering to device NVIDIA GeForce RTX 3080
Creating D3D11 device
Found adapter for XR rendering: NVIDIA GeForce RTX 3080
XR creating session failed: XR_ERROR_RUNTIME_FAILURE (-2)
XR creating seated reference space failed: XR_ERROR_HANDLE_INVALID (-12)

I have two vr rigs, so will endeavour to find time to try my second one.

edit: completely removed my custom openxr install and installed the latest microsoft one.  Same errors.

edit2: tried it on my second vr rig with a completely different wmr headset, same errors.

Cheers,
divxmaster

**Comment by divxmaster1** [comment]
Update: did a fresh windows reinstall of my main vr rig. It now works!   I need to set "use latest preview openxr runtime" to on in openxr tools, to make it work in direct openxr mode (due to my CAS/OC hack mentioned above).  It has a bindings error in this mode though, but steamvr->openxr mode seems to work.

Cheers,
divxmaster
